crates/nu-parser/src/parser.rs

```diff
@@ -222,15 +221,186 @@ pub(crate) fn check_call(
     }
 }
 
+/// Parses a string in the arg or head position of an external call.
+///
+/// If the string begins with `r#`, it is parsed as a raw string. If it doesn't contain any quotes
+/// or parentheses, it is parsed as a glob pattern so that tilde and glob expansion can be handled
+/// by `run-external`. Otherwise, we use a custom state machine to put together an interpolated
+/// string, where each balanced pair of quotes is parsed as a separate part of the string, and then
+/// concatenated together.
+///
+/// For example, `-foo="bar\nbaz"` becomes `$"-foo=bar\nbaz"`
```

Comment:
Is the `$` necessary here? Wouldn't it be the same as `"-foo=bar\nbaz"`? Or do you do also actual string interpolation, e.g., `-foo=($bar)` becoming `$"-foo=($bar)"`?

Reply:
This comment might actually not be accurate because I remember changing it to not be string interpolation if every component ends up being a string 

But in general the idea was that because each piece of the bareword could either have quotes (so we don't handle interpolation) or not (so we do) then the whole thing needs to be an interpolation

Reply:
OK, so the consequence of this is the following (just tested):
```
> let foo = "spam"
> ^echo --foo=($spam)
--foo=spam
```
which I think is pretty cool, I'm just not sure whether it was intended :laughing:.

Reply:
it was intentional! I figured that would be nice to be able to do, and it's more consistent